sklearn/utils/estimator_checks.py

```diff
@@ -643,6 +644,48 @@ def check_fit2d_predict1d(name, estimator_orig):
                                  getattr(estimator, method), X[0])
 
 
+@ignore_warnings(category=(DeprecationWarning, FutureWarning))
+def check_methods_subset_invariance(name, estimator_orig):
+    # check that method gives invariant results if applied
+    # one by one or on all elements together.
+    rnd = np.random.RandomState(0)
+    X = 3 * rnd.uniform(size=(20, 3))
+    X = pairwise_estimator_convert_X(X, estimator_orig)
+    y = X[:, 0].astype(np.int)
+    estimator = clone(estimator_orig)
+    y = multioutput_estimator_convert_y_2d(estimator, y)
+
+    if hasattr(estimator, "n_components"):
+        estimator.n_components = 1
+    if hasattr(estimator, "n_clusters"):
+        estimator.n_clusters = 1
+
+    set_random_state(estimator, 1)
+    estimator.fit(X, y)
+
+    for method in ["predict", "transform", "decision_function",
+                   "score_samples", "predict_proba"]:
+        if hasattr(estimator, method):
+            msg = ("{method} of {name} is not invariant when applied "
+                   "to a subset.").format(method=method, name=name)
+            func = getattr(estimator, method)
+            res_all = func(X)
+            res_one = [func(X[i].reshape(1, X.shape[1]))
+                       for i in range(X.shape[0])]
+            # func can output tuple (e.g. score_samples)
+            if type(res_all) == tuple:
+                res_all = res_all[0]
+                res_one = list(map(lambda x: x[0], res_one))
+            # TODO remove cases when corrected
+            if [name, method] in [['SVC', 'decision_function'],
```

Comment:
use tuples. Like `if name, method in [('SVC', 'decision_function'), ...]`

tuples are intended for struct-like objects where each field means a different thing. Arrays are usually for homogeneous semantics